### PR TITLE
Improve changelog design

### LINKED ISSUE #7252: Improve changelog design
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation
The changelog needs a redesign because:
- Low readability
- Duplicate wordings (fix..., fix..., fix...)

### Feature / Enhancement Description
A design similar to the [Node.js changelog](https://github.com/nodejs/node/blob/master/CHANGELOG.md).

### Example Use Case
(n/a)

### Alternatives / Workarounds
(n/a)

### 3rd Party References
- [Node.js changelog](https://github.com/nodejs/node/blob/master/CHANGELOG.md)

### PR DESCRIPTION
**Pull request by mtrezza**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description
Changelog redesign

Related issue: closes #7252

### Approach
Improved design inspired by [Node.js changelog](https://github.com/nodejs/node/blob/master/CHANGELOG.md).

- Added a version table
- Condensed entry by removing link to contributor, instead letting the contributor add their GitHub handle or real name as plain text; GitHub handles are not persistent anyway and break if the user changes their name, see [docs](https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/changing-your-github-username#about-username-changes); new entry format:
   > [Description] ([Contributor]) #[Link to PR]
   > Added something new (David Hume) [#1711](#)
- Release section contains following headlines:
  - Breaking Changes
    - Any change that is breaking, regardless of what the change is about; for example it can be a bug fix or a new feature that replaces old functionality
  - Notable Changes
    - New important or useful features or improvements, but never bug fixes, unless published security vulnerabilities.
    - Security vulnerabilities should be mentioned with their CVS ID.
  - ~~Bug Fixes~~
    - ~~All bug fixes, unless it's a breaking fix, then it goes under "Breaking Changes".~~
  - Other Commits
    - Any minor changes, small improvement, bug fixes (unless published security vulnerability).

### TODOs before merging

(none)

### CODE REVIEW

**Comment by cbaker6** [comment]
I like it! Easy to navigate and easy to search for notable changes

**Comment by mtrezza** [comment]
Do you think we should remove the "Bug Fixes" section and just add them to "Other Commits"? That would simplify and I am not sure what we would want to distinguish bug fixes.

**Comment by TomWFox** [comment]
Some much needed TLC, I really like the navigation!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,50 +1,128 @@
...
-    - NEW
-    - IMPROVE
-    - FIX
-    Thanks for contributing to Parse Server!
--->
-
-### master
-[Full Changelog](https://github.com/parse-community/parse-server/compare/4.5.0...master)
+# Parse Server Changelog
+## Current Major Version
```
Comment (TomWFox) [inline CHANGELOG.md:2]: My only nit is this heading - it seems a bit verbose to me & visually a bit much.

Comment (mtrezza) [reply CHANGELOG.md:2]: Which one

**Comment by mtrezza** [comment]
I have merged the "Bug Fixes" into "Other Commits". The separate section does not seem to provide any additional value for readers, because:
- A bug fix per se does not say anything about the importance or relevance of the change
- If the bug fix has high relevance, it can go under "Notable Changes"
- The distinction between bug fix and a "common change" seems to be not clear, so why bother trying to categorize that

File: CHANGELOG.md (hunk 1 above)
Comment (TomWFox) [reply CHANGELOG.md]: Yes I like that 👌

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,50 +1,126 @@
...
+- Test Parse Server continuously against all recent Node.js versions that have not reached their end-of-life support date, added Node.js compatibility table to Parse Server docs (Manuel Trezza) [7161](https://github.com/parse-community/parse-server/pull/7177)
+- Throw error on invalid Cloud Function validation configuration (dblythy) [#7154](https://github.com/parse-community/parse-server/pull/7154)
+- Allow Cloud Validator `options` to be async (dblythy) [#7155](https://github.com/parse-community/parse-server/pull/7155)
+- Optimize queries on classes with pointer permissions (Pedro Diaz) [#7061](https://github.com/parse-community/parse-server/pull/7061)
+- Test Parse Server continuously against all relevant Postgres versions (minor versions), added Postgres compatibility table to Parse Server docs (Corey Baker) [#7176](https://github.com/parse-community/parse-server/pull/7176)
+- Allow Cloud Validator `options` to be async (dblythy) [#7155](https://github.com/parse-community/parse-server/pull/7155)
```
Comment (mtrezza) [inline CHANGELOG.md:117]: Duplicate, remove

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,50 +1,126 @@
...
-- FIX: Winston Logger interpolating stdout to console [#7114](https://github.com/parse-community/parse-server/pull/7114). Thanks to [dplewis](https://github.com/dplewis)
-- FIX: Move graphql-tag from devDependencies to dependencies. [7183](https://github.com/parse-community/parse-server/pull/7183). Thanks to [Antonio Davi Macedo Coelho de Castro](https://github.com/davimacedo).
-
-### 4.5.0
+## Unreleased (Master Branch)
+[Full Changelog](https://github.com/parse-community/parse-server/compare/4.5.0...master)
+### Breaking Changes
+- Added file upload restriction. File upload is now only allowed for authenticated users by default for improved security. To allow file upload also for Anonymous Users or Public, set the `fileUpload` parameter in the [Parse Server Options](https://parseplatform.org/parse-server/api/master/ParseServerOptions.html) (dblythy, Manuel Trezza) [#7071](https://github.com/parse-community/parse-server/pull/7071)
+### Notable Changes
+- Added Parse Sever Security Check to report weak security settings (Manuel Trezza, dblythy) [#7247](https://github.com/parse-community/parse-server/issues/7247)
```
Comment (dblythy) [inline CHANGELOG.md:94]: typo on 'sever' (server)

File: CHANGELOG.md [hunk 4]
```diff
@@ -456,66 +532,66 @@ Security Fix: see Advisory: [GHSA-2479-qvv7-47q](https://github.com/parse-commun
...
 * Fixes issue that prevented ACL's from being used with `select` (see [#571](https://github.com/parse-community/Parse-SDK-JS/issues/571))
 
-#### Dependency updates:
+### Dependency updates:
 * [@parse/simple-mailgun-adapter@1.1.0](https://www.npmjs.com/package/@parse/simple-mailgun-adapter)
 * [mongodb@3.1.3](https://www.npmjs.com/package/mongodb)
 * [request@2.88.0](https://www.npmjs.com/package/request)
 
-##### Devevelopment Dependencies Updates:
+### Devevelopment Dependencies Updates:
```
Comment (dblythy) [inline CHANGELOG.md:594]: probably doesn't matter but typo on "Devevelopment"

File: CHANGELOG.md [hunk 5]
```diff
@@ -96,7 +172,7 @@ ___
 - IMPROVE: Optimized deletion of class field from schema by using an index if available to do an index scan instead of a collection scan. [#6815](https://github.com/parse-community/parse-server/issues/6815). Thanks to [Manuel Trezza](https://github.com/mtrezza).
 - IMPROVE: Enable MongoDB transaction test for MongoDB >= 4.0.4 [#6827](https://github.com/parse-community/parse-server/pull/6827). Thanks to [Manuel](https://github.com/mtrezza).
 
-### 4.3.0
+## 4.3.0
 [Full Changelog](https://github.com/parse-community/parse-server/compare/4.2.0...4.3.0)
 - PERFORMANCE: Optimizing pointer CLP query decoration done by DatabaseController#addPointerPermissions [#6747](https://github.com/parse-community/parse-server/pull/6747). Thanks to [mess-lelouch](https://github.com/mess-lelouch).
 - SECURITY: Fix security breach on GraphQL viewer [78239ac](https://github.com/parse-community/parse-server/commit/78239ac9071167fdf243c55ae4bc9a2c0b0d89aa), [secuity advisory](https://github.com/parse-community/parse-server/security/advisories/GHSA-236h-rqv8-8q73). Thanks to [Antoine Cormouls](https://github.com/Moumouls).
```
Comment (dblythy) [inline CHANGELOG.md:178]: typo on "secuity" (security)

**Comment by dblythy** [comment]
This looks really good.

Would it perhaps still be worth having a "new features" section, as well as the notable changes? When I read the CHANGELOG I like looking for new and cool things I can implement on my Parse Server.

**Comment by mtrezza** [comment]
Yes, I sugggest `Notable` rather than `New` because something can be new but relevant only for 1% of deployments, while something can be an improvement but very relevant. This gives us more versatility.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,50 +1,125 @@
...
+[Full Changelog](https://github.com/parse-community/parse-server/compare/4.5.0...master)
+### Breaking Changes
+- Added file upload restriction. File upload is now only allowed for authenticated users by default for improved security. To allow file upload also for Anonymous Users or Public, set the `fileUpload` parameter in the [Parse Server Options](https://parseplatform.org/parse-server/api/master/ParseServerOptions.html) (dblythy, Manuel Trezza) [#7071](https://github.com/parse-community/parse-server/pull/7071)
+### Notable Changes
+- Added Parse Server Security Check to report weak security settings (Manuel Trezza, dblythy) [#7247](https://github.com/parse-community/parse-server/issues/7247)
+- EXPERIMENTAL: Added new page router with placeholder rendering and localization of custom and feature pages such as password reset and email verification (Manuel Trezza) [#6891](https://github.com/parse-community/parse-server/issues/6891)
+- EXPERIMENTAL: Added custom routes to easily customize flows for password reset, email verification or build entirely new flows (Manuel Trezza) [#7231](https://github.com/parse-community/parse-server/issues/7231)
+### Other Commits
```
Comment (TomWFox) [inline CHANGELOG.md:97]: Apologies for only bringing this up now...

How about changing this to `### Other changes`? -  1) in keeping with the other headings 2) `other commits` could imply all other commits, when in reality we tend to omit CI changes, readme changes etc

Comment (mtrezza) [reply CHANGELOG.md:97]: Yes, makes sense, I'll change that with one of the next PRs instead of opening a new one.

**Comment by TomWFox** [comment]
Personally I would also have put some more of the changes in the notable section...

```
- request.context for afterFind triggers (dblythy) [#7078](https://github.com/parse-community/parse-server/pull/7078)
- LiveQuery support for $and, $nor, $containedBy, $geoWithin, $geoIntersects queries (dplewis) [#7113](https://github.com/parse-community/parse-server/pull/7113)
- Added `requireAnyUserRoles` and `requireAllUserRoles` for Parse Cloud validator (dblythy) [#7097](https://github.com/parse-community/parse-server/pull/7097)
- Added new account lockout policy option `accountLockout.unlockOnPasswordReset` to automatically unlock account on password reset (Manuel Trezza) [#7146](https://github.com/parse-community/parse-server/pull/7146)
- Allow Cloud Validator `options` to be async (dblythy) [#7155](https://github.com/parse-community/parse-server/pull/7155)
```

What do you think?

File: CHANGELOG.md (hunk 6 above)
Comment (mtrezza) [reply CHANGELOG.md:97]: Changed in https://github.com/parse-community/parse-server/pull/7272

**Comment by mtrezza** [comment]
The suggested purpose of `Notable Changes` is to have a list of significant changes, like an entirely new feature, or a significant improvement. Like, if we sold Parse Server in a box and had to design the front, what would it say? Probably not that a method is now async or accepts an additional parameter. I think we should keep a high entry barrier for that section, if we populate it inflationary, we will end up with only CI changes and bug fixes in the `Other Changes`. Every change is important for someone after all.

So from all of the changes you mentioned above, I personally don't see any passing that criteria, but I am open to other suggestions of how to use the `Notable Changes` section.

File: CHANGELOG.md (hunk 6 above)
Comment (TomWFox) [reply CHANGELOG.md:97]: Thanks 👍

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0](https://github.com/parse-community/parse-server/releases/tag/5.0.0)

**Pull request merged**